src/ServicePulse.Host/vue/src/components/DataView.vue

```diff
@@ -0,0 +1,173 @@
+<script setup>
+import { computed, onMounted, onUnmounted, ref } from "vue";
+import { useFetchFromServiceControl } from "../composables/serviceServiceControlUrls";
+
+const props = defineProps({
+  apiUrl: String,
+  itemsPerPageOptions: {
+    type: Array,
+    default: () => [20, 35, 50, 75],
+  },
+  itemsPerPage: {
+    type: Number,
+    default: 50,
+  },
+  autoRefresh: {
+    type: Number,
+    default: 0,
+  },
+  showPagination: {
+    type: Boolean,
+    default: true,
+  },
+  showItemsPerPage: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+const refreshTimer = ref();
+const items = ref([]);
+const pageNumber = ref(1);
+const itemsPerPage = ref(props.itemsPerPage);
+const totalCount = ref(0);
+const showPagination = computed(() => {
+  return props.showPagination && numberOfPages.value > 1;
+});
+
+const numberOfPages = computed(() => {
+  return Math.ceil(totalCount.value / itemsPerPage.value);
+});
+
+const showItemsPerPage = computed(() => {
+  return props.showItemsPerPage;
+});
+
+function changeItemsPerPage(value) {
+  itemsPerPage.value = value;
+  loadData();
+}
+
+function calculatePageNumbers() {
```

Comment:
I feel that this function should contain _all_ the logic for determining page numbers, and return an array of objects { label: (number or ...), setPage: function } which the template can do a single iteration over

Reply:
Done